lib/valid_email2.rb

```diff
@@ -1,24 +1,34 @@
 # frozen_string_literal: true
 
 require "valid_email2/email_validator"
+require_relative "./helpers/deprecation_helper"
 
 module ValidEmail2
+  
   BLACKLIST_FILE  = "config/blacklisted_email_domains.yml"
+  DENY_LIST_FILE  = "config/deny_listed_email_domains.yml"
   WHITELIST_FILE  = "config/whitelisted_email_domains.yml"
+  ALLOW_LIST_FILE  = "config/allow_listed_email_domains.yml"
   DISPOSABLE_FILE = File.expand_path('../config/disposable_email_domains.txt', __dir__)
 
   class << self
+    extend DeprecationHelper
+
     def disposable_emails
       @disposable_emails ||= load_file(DISPOSABLE_FILE)
     end
 
     def blacklist
-      @blacklist ||= load_if_exists(BLACKLIST_FILE)
+      @deny_list ||= load_if_exists(DENY_LIST_FILE || BLACKLIST_FILE)
```

Comment:
> DENY_LIST_FILE || BLACKLIST_FILE

This change made under the assumption old users would have a blacklist file while new users will have only an deny-list file so we need to check which they have.

I'm not entirely sure if the intention behind the `config/blacklisted_email_domains.yml` and config/`whitelisted_email_domains.yml` files is for users to be able to update these on their own, but if that's the case we'll probably need to add something to the README saying the user should update the name (i.e., config/whitelist -> config/allow_list) or contents of their files, or write a script to do so for them automatically.

Reply:
I'd go just for the README and let the user handle it, no need for a script

Reply:
Unless i'm missing something, i don't think this does what you want it to do, it will never use the blacklist file as `DENY_LIST_FILE` will always be truthy and we will never reach `BLACKLIST_FILE`.

I think i would like to solve this in a way that lets users with the old filename be notified that they need to change the name of the file.
My fear is that there are users that have set the gem up with a blacklist file and forgotten about it, so i think we are best of doing them the service of notifying them that they need to change the name of the file.

---

The intention is to allow the users to allow or deny -list domains if they need, without having to open a PR to have it updated in the gem.

Reply:
> Unless i'm missing something, i don't think this does what you want it to do, it will never use the blacklist file as DENY_LIST_FILE will always be truthy and we will never reach BLACKLIST_FILE.

Ah, I see. Thanks for pointing that out. I was operating under the assumption a user would have either a `DENY_LIST_FILE` or a `BLACKLIST_FILE`, but not both.

> I think i would like to solve this in a way that lets users with the old filename be notified that they need to change the name of the file.

Good call, and I like the way you handled this in 46634dff409f9a47f10684d629dc022b22daf362.